engine/pokemon/summary_screen.asm

```diff
@@ -0,0 +1,1162 @@
+
+	ld hl, wSummaryScreenTypes
+	ld a, [wBaseType1]
+	ld [hli], a
+	ld a, [wBaseType2]
+	ld [hl], a
+
+	lb bc, 0, 252
+.moveTypes
+	ld hl, wTempMonMoves - 252
+	add hl, bc
+	ld a, [hl]
+	and a
+	jr z, .moveTypesEnd
+	ld [wCurMove], a
+	ld hl, Moves + MOVE_TYPE
+	call GetCurMoveProperty
+	ld hl, wSummaryScreenTypes - 252 + 2
+	add hl, bc
+	ld [hl], a
+	inc c
+	jr nz, .moveTypes
+.moveTypesEnd
+
+	call .ApplySummaryPalettes
+
+	hlcoord 19 - NUM_SUMMARY_PAGES * 2, 0
+	ld [hl], SUMMARY_TILE_LEFT_ARROW
+	hlcoord 19, 0
+	ld [hl], SUMMARY_TILE_RIGHT_ARROW
+	ret
+
+.PlaceHPBar:
+	ld hl, wTempMonHP
+	ld a, [hli]
+	ld b, a
+	ld c, [hl]
+	ld hl, wTempMonMaxHP
+	ld a, [hli]
+	ld d, a
+	ld e, [hl]
+	farcall ComputeHPBarPixels
+	ld hl, wCurHPPal
+	call SetHPPal
+	jmp DelayFrame
+
+.ApplySummaryPalettes:
+	ldh a, [rSVBK]
+	push af
+	ld a, BANK(wOBPals1)
+	ldh [rSVBK], a
+	ld c, 0
+.objPalettes
+	ld d, 0
+	ld hl, wSummaryScreenTypes
+	add hl, bc
+	ld e, [hl]
+	sla e
+	ld hl, TypeIconPals
+	add hl, de
+	push hl
+	ld hl, wOBPals1 + 6
+	add hl, bc
+	add hl, bc
+	add hl, bc
+	add hl, bc
+	add hl, bc
+	add hl, bc
+	add hl, bc
+	add hl, bc
+	ld d, h
+	ld e, l
+	pop hl
+	push bc
+	farcall LoadOneColor
+	pop bc
+	inc c
+	ld a, 5
+	cp a, c
+	jr nc, .objPalettes
+
+	farcall ApplyOBPals
+	pop af
+	ldh [rSVBK], a
+	ret
+
+.PageSprites:
+	db 17, 104, SUMMARY_TILE_OAM_PAGE, $0
+	db 17, 120, SUMMARY_TILE_OAM_PAGE, $1
+	db 17, 136, SUMMARY_TILE_OAM_PAGE, $2
+	db 17, 152, SUMMARY_TILE_OAM_PAGE, $3
+
+.TabTitleSprites:
+	db 108, 20, SUMMARY_TILE_OAM_TITLES + 0, $0
+	db 108, 28, SUMMARY_TILE_OAM_TITLES + 1, $0
+	db 108, 36, SUMMARY_TILE_OAM_TITLES + 2, $0
+	db 108, 44, SUMMARY_TILE_OAM_TITLES + 3, $0
+
+SummaryScreen_InitAttrmap:
+	farcall WipeAttrMap
+
+	hlcoord 0, 0, wAttrmap
+	lb bc, 12, 8
+	ld a, SUMMARY_PAL_POKEMON
+	call FillBoxWithByte
+
+	hlcoord 8, 0, wAttrmap
+	lb bc, 1, 14
+	; a == SUMMARY_PAL_POKEMON
+	call FillBoxWithByte
+
+	hlcoord 0, 12, wAttrmap
+	lb bc, 1, 20
+	ld a, SUMMARY_PAL_SIDE_WINDOW
+	call FillBoxWithByte
+
+	hlcoord 2, 12, wAttrmap
+	lb bc, 1, 3
+	xor a
+	assert SUMMARY_PAL_LOWER_WINDOW == 0
+	call FillBoxWithByte
+
+	hlcoord 1, 11, wAttrmap
+	ld a, SUMMARY_PAL_SIDE_WINDOW
+	ld [hli], a
+	ld a, Y_FLIP | SUMMARY_PAL_LOWER_WINDOW
+	ld bc, 3
+	rst ByteFill
+	ld a, X_FLIP | SUMMARY_PAL_SIDE_WINDOW
+	ld [hl], a
+	hlcoord 5, 12, wAttrmap
+	ld [hl], a
+
+	hlcoord 7, 1, wAttrmap
+	lb bc, 11, 13
+	ld a, SUMMARY_PAL_SIDE_WINDOW
+	call FillBoxWithByte
+
+	hlcoord 5, 9, wAttrmap
+	ld [hl], SUMMARY_PAL_GENDER_MARKER
+
+	hlcoord 2, 8, wAttrmap
+	lb bc, 3, 3
+	ld a, SUMMARY_PAL_ITEM
+	call FillBoxWithByte
+	ret
+
+SummaryScreen_LoadPage:
+	ld a, [wCurPartySpecies]
+	ld [wTempSpecies], a
+	ld [wCurSpecies], a
+	xor a
+	ldh [hBGMapMode], a
+	ldh [hOAMUpdate], a
+	ldh [hCGBPalUpdate], a
+	call .ClearBox
+	call .PlaceLevelAndGender
+	hlbgcoord 0, 0, wSummaryScreenWindowBuffer
+	ld a, 10
+	ld e, a
+	ld b, 0
+.loop
+	ld c, $10
```

Comment:
what is $10?

Reply:
ByteFill amount, b is cleared before the loop and remains cleared